### PR TITLE
endpoint can be overridden with env var

### PR DESCRIPTION
**Pull request by cob16**
- removes hardcoded URLs from the gateway

- allow env var OPTICS_ENDPOINT to overrite endpoint
- if OPTICS_ENDPOINT is not provieded default to the uat enviroment

### CODE REVIEW

File: app/lib/gateway/optics.rb [hunk 1]
```diff
@@ -2,8 +2,10 @@ module Gateway
   class Optics
     class ClientError < StandardError; end
 
-    GET_TOKEN_ENDPOINT = 'https://uat.icasework.com/token?db=hmcts'.freeze
-    ENDPOINT = 'https://uat.icasework.com/createcase?db=hmcts'.freeze
+    def initialize(endpoint:)
+      @get_token_url = "#{endpoint}/token?db=hmcts".freeze
```
Comment (tommotaylor) [inline app/lib/gateway/optics.rb]: Seems odd to have half the endpoint in ENV vars and half of it hardcoded here. Can't we have it all as an env var?

Comment (cob16) [reply app/lib/gateway/optics.rb]: the reasoning was that the URL will never change but the endpoint will change to each env. having the full URLs might increase the chance for error

Comment (tommotaylor) [reply app/lib/gateway/optics.rb]: URLs might change.

Comment (cob16) [reply app/lib/gateway/optics.rb]: In my option removing visibility of what the true path requests are being made to is will make the code less understandable thats why a singe endpoint var was desireable. 

So we want 2 separate ENV vars? 
1 for auth and a fully separate one for posting?

Comment (tommotaylor) [reply app/lib/gateway/optics.rb]: Yeah thats a fair point

**Pull request merged**